include/exadg/postprocessor/time_control.h

```diff
@@ -0,0 +1,99 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_COMPRESSIBLE_NAVIER_STOKES_POSTPROCESSOR_TIME_CONTROL_H_
+#define INCLUDE_COMPRESSIBLE_NAVIER_STOKES_POSTPROCESSOR_TIME_CONTROL_H_
+
+/**
+ *This class provides information at which time steps a certain action should be triggered. It is
+ *typically used in postprocessing routines that shall not be invoked after every time step. The
+ *user can specify an evaluation interval in terms of number of time steps or an amount of time.
+ */
+
+// deal.ii
+#include <deal.II/base/conditional_ostream.h>
+
+// ExaDG
+#include <exadg/utilities/numbers.h>
+
+namespace ExaDG
+{
+struct TimeControlData
+{
+  TimeControlData();
+
+  bool         is_active;
+  double       start_time;
+  double       end_time;
+  double       trigger_interval;
+  unsigned int trigger_every_time_steps;
+  unsigned int trigger_every_sub_time_step;
+  bool         force_final_evaluation;
+  unsigned int counter_start;
```

Comment:
I believe these parameters three should not be part of `TimeControlData`

Reply:
I am currently not sure about `force_final_evaluation`. The problem with `counter_start` is that the counter is reset automatically in case of a restart for example. So I see a redundancy here. As written elsewhere, `sub_time_step` is related to statistical averaging and should be realized somewhere else in my opinion.

Reply:
I think expressing tolerance in terms of ```end_time - start_time``` will not help since the critical value here is ```time_step_size```. We could cache ```last_time``` and determine the ```time_step_size``` with this. 

If this is not an argument for enforce_final_evaluation, which purpose has the variable?

Reply:
maybe also `time_step_size`: `if(time + 0.01*time_step_size > end_time) -> needs_evaluation = true;`

Reply:
But is this realistic? Say a time step size cannot be solved faster than t_wall,dt=1e-5 (hardware limits), then such a simulation with 1e12 time steps still runs t_wall=1e7 seconds, or 115 days! I am willing to find a better solution ...

Reply:
Let's keep it a fixed tolerance and address it some other time